wgpu-hal/src/dx12/mod.rs

```diff
@@ -221,7 +221,7 @@ pub struct Device {
     private_caps: PrivateCapabilities,
     shared: Arc<DeviceShared>,
     // CPU only pools
-    rtv_pool: Mutex<descriptor::CpuPool>,
+    rtv_pool: Arc<Mutex<descriptor::CpuPool>>,
```

Comment:
I don't know if `Arc::try_unwrap(self.shared).ok().map(|shared| shared.rtv_pool.into_inner().destroy())` is the right way to destroy rtv_pool

Reply:
I brought this up [in Matrix](https://matrix.to/#/!FZyQrssSlHEZqrYcOb:matrix.org/$CP4QMBU4CGgAbv6bigj6XBohhf1Uv9N_3pm6JuxhENU?via=matrix.org&via=mozilla.org&via=kde.org) and the suggestion there was that we create the null descriptor when we create the Device, let it get destroyed when the Device is destroyed, and then just give every CommandEncoder a copy of the handle. That way, there's no need to allocate the null descriptor in `begin_render_pass`, so there's no need for `rtv_pool` to be shared.

In chat, @magcius (jasperrlz) said:
> RTV/DSV descriptors are weird, because they don't have to persist in a pool. As long as the descriptors are alive at the time of bind, it's fine.